src/newgrf.cpp

```diff
@@ -658,12 +653,8 @@ static Engine *GetNewEngine(const GRFFile *file, VehicleType type, uint16 intern
 	e->grf_prop.grffile = file;
 
 	/* Reserve the engine slot */
-	assert(_engine_mngr.Length() == e->index);
-	EngineIDMapping *eid = _engine_mngr.Append();
-	eid->type            = type;
-	eid->grfid           = scope_grfid; // Note: this is INVALID_GRFID if dynamic_engines is disabled, so no reservation
-	eid->internal_id     = internal_id;
-	eid->substitute_id   = min(internal_id, _engine_counts[type]); // substitute_id == _engine_counts[subtype] means "no substitute"
+	assert(_engine_mngr.size() == e->index);
+	_engine_mngr.push_back({scope_grfid, internal_id, static_cast<byte>(type), static_cast<uint8>(min(internal_id, _engine_counts[type]))});
```

Comment:
`src/newgrf.cpp:657:52: error: suggest braces around initialization of subobject [-Werror,-Wmissing-braces]`

Reply:
Oh yeah, that reminds me of my next crusade, enums. Not sure what to do about this one